lib/experimental/blocks.php

```diff
@@ -35,24 +35,25 @@ function build_comment_query_vars_from_block( $block ) {
 			$comment_args['hierarchical'] = false;
 		}
 
-		$per_page     = get_option( 'comments_per_page' );
-		$default_page = get_option( 'default_comments_page' );
-
-		if ( $per_page > 0 ) {
-			$comment_args['number'] = $per_page;
-
-			$page = (int) get_query_var( 'cpage' );
-			if ( $page ) {
-				$comment_args['paged'] = $page;
-			} elseif ( 'oldest' === $default_page ) {
-				$comment_args['paged'] = 1;
-			} elseif ( 'newest' === $default_page ) {
-				$comment_args['paged'] = (int) ( new WP_Comment_Query( $comment_args ) )->max_num_pages;
-			}
-			// Set the `cpage` query var to ensure the previous and next pagination links are correct
-			// when inheriting the Discussion Settings.
-			if ( 0 === $page && isset( $comment_args['paged'] ) && $comment_args['paged'] > 0 ) {
-				set_query_var( 'cpage', $comment_args['paged'] );
+		if ( get_option( 'page_comments' ) === '1' ) {
```

Comment:
I'm not an expert here so I might be wrong, but this would fail if `page_comments` were `true` instead of `1`. Should we not use the `==` (double equality) operator instead?

Reply:
Yep, as far as I've checked `page_comments` saves as 1 or 0 string in the database (available at http://wordpresssite.com/wp-admin/options.php). But I guess that will be safer if we use double equality.